### PR TITLE
Improve issue templates

### PR DESCRIPTION
**Pull request by stephsamson**
Based heavily off of the issue templates of [spaCy](https://github.com/explosion/spaCy/issues/new/choose) which directs support questions to StackOverflow under the tag `python-poetry` [link](https://stackoverflow.com/questions/tagged/python-poetry). This should hopefully make it easier for contributors to triage bug fixes and support questions.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/---bug-report.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 ---
-name: Bug report
-about: Report errors and problems
+name: "\U0001F41E Bug Report"
+about: Did you find a bug?
+title: ''
+labels: ''
```
Comment (brycedrennan) [inline .github/ISSUE_TEMPLATE/---bug-report.md]: any reason to not add the bug label here? I assume that will auto-add the label to the issue?

File: .github/ISSUE_TEMPLATE/---documentation.md [hunk 2]
```diff
@@ -1,6 +1,9 @@
 ---
-name: Documentation issue
-about: Report errors and problems with the documentation (https://poetry.eustace.io/docs/)
+name: "\U0001F4DA Documentation"
+about: Did you find errors, problems, or anything unintelligible in the docs?
+title: ''
```
Comment (brycedrennan) [inline .github/ISSUE_TEMPLATE/---documentation.md]: I think the link to the docs is useful?

File: .github/ISSUE_TEMPLATE/---everything-else.md [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+about: For questions and issues that do not fall in any of the other categories. This
+  can include questions about Poetry's roadmap. For support questions, please post
+  on StackOverflow.
+title: ''
+labels: ''
+assignees: ''
+
+---
+
+<!-- Describe your question and issue here. This space is meant to be used for general questions that are neither bugs, feature requests, nor documentation issues. A good example would be a question regarding Poetry's roadmap, for example. If you're looking for help when it comes to using Poetry, consider posting a question on StackOverflow instead: http://stackoverflow.com/questions/tagged/python-poetry -->
```
Comment (brycedrennan) [inline .github/ISSUE_TEMPLATE/---everything-else.md]: Nice!

File: .github/ISSUE_TEMPLATE/---bug-report.md (hunk 1 above)
Comment (stephsamson) [reply .github/ISSUE_TEMPLATE/---bug-report.md]: Ah, good catch, thanks.

File: .github/ISSUE_TEMPLATE/---documentation.md (hunk 2 above)
Comment (stephsamson) [reply .github/ISSUE_TEMPLATE/---documentation.md]: Indeed, I'll add it back.

**Pull request merged**